crates/uv/src/commands/tool/install.rs

```diff
@@ -125,6 +125,7 @@ pub(crate) async fn install(
     let name = from.name.to_string();
 
     let installed_tools = InstalledTools::from_settings()?;
+    let _lock = installed_tools.acquire_lock()?;
```

Comment:
@charliermarsh and here the `init` is deferred all the way to [L305](https://github.com/astral-sh/uv/pull/4720/files#diff-0dfdec55a014c8bde81440471c7410d26a840618e095bdb007abedf8866e51e3R305)